fix(core): Deprecate TestBed.get(...):any

Goodwine opened the pull request:
Adds an overload to TestBed.get making parameters strongly typed and
deprecated previous signature that accepted types `any`. The function
still returns `any` to prevent build breakages, but eventually stronger
type checks will be added so a future Angular version will break builds
due to additional type checks.
See previous breaking change - #13785

Issue #26491

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
```ts
get(token: any, default?: any): any
```

Issue Number: #26491


## What is the new behavior?
"Same" signature as `Injector.get`:
```ts
get(token: InjetionToken<T>|Type<T>, default?: T, flags?: InjectorFlags): any
```
(Instead of returning `T`, it still returns `any`) to prevent breaking changes for 8.0, but this will hopefully break on 9.0

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

Note that the return type is still `any`, this will change to be an actual breaking change similar to #13785 in a future major version

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

Goodwine commented:
Note: At head these tests were already failing -

//packages/compiler-cli/test:extract_i18n
//packages/compiler-cli/test:ngc
//packages/compiler-cli/test:ngtools_api
//packages/compiler-cli/test:perform_compile
//packages/compiler-cli/test:perform_watch
//packages/compiler-cli/test/diagnostics:check_types
//packages/compiler-cli/test/diagnostics:expression_diagnostics
//packages/compiler-cli/test/transformers:test
//packages/core/test/bundling/cyclic_import:symbol_test
//packages/core/test/bundling/cyclic_import:test
//packages/core/test/bundling/hello_world:symbol_test
//packages/core/test/bundling/hello_world:test
//packages/core/test/bundling/todo:symbol_test
//packages/core/test/bundling/todo:test
//packages/core/test/bundling/todo_i18n:test
//packages/core/test/bundling/todo_r2:test
//packages/language-service/test:test

mhevery commented on packages/core/testing/src/r3_test_bed.ts:
```diff
@@ -10,52 +10,12 @@
-} from '@angular/core';
 // clang-format on
 import {ResourceLoader} from '@angular/compiler';
+import {ApplicationInitStatus, COMPILER_OPTIONS, Compiler, CompilerOptions, Component, Directive, ErrorHandler, InjectFlags, InjectionToken, Injector, ModuleWithComponentFactories, NgModule, NgModuleFactory, NgZone, Pipe, PlatformRef, Provider, SchemaMetadata, StaticProvider, Type, resolveForwardRef, ɵInjectableDef as InjectableDef, ɵNG_COMPONENT_DEF as NG_COMPONENT_DEF, ɵNG_DIRECTIVE_DEF as NG_D ...
```
Why did you reformat this? Can you revert?

Goodwine replied:
This changed because 2 types were added (InjectFlags, InjectionToken) and the format script updated this "changed lines", a recent patch added clang-off so I guess it's not changing anymore (?)

Anyways, done :)

Goodwine committed:
fix(core): Deprecate TestBed.get(...):any

Adds an overload to TestBed.get making parameters strongly typed and
deprecated previous signature that accepted types `any`. The function
still returns `any` to prevent build breakages, but eventually stronger
type checks will be added so a future Angular version will break builds
due to additional type checks.
See previous breaking change - #13785

Issue #26491

Goodwine commented:
http://test/OCL:241762158:BASE:241762792:1554313801393:5a80dbe6
